example/sparse/linear_model.py

```diff
@@ -0,0 +1,34 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import mxnet as mx
+from weighted_softmax_ce import *
+
+def linear_model(num_features, positive_cls_weight):
+    # data with csr storage type to enable feeding data with CSRNDArray
+    x = mx.symbol.Variable("data", stype='csr')
+    norm_init = mx.initializer.Normal(sigma=0.01)
+    # weight with row_sparse storage type to enable sparse gradient updates
+    weight = mx.symbol.Variable("weight", shape=(num_features, 2),
+                                init=norm_init, stype='row_sparse')
+    bias = mx.symbol.Variable("bias", shape=(2,))
+    dot = mx.symbol.sparse.dot(x, weight)
+    pred = mx.symbol.broadcast_add(dot, bias)
+    y = mx.symbol.Variable("softmax_label")
+    model = mx.sym.Custom(pred, y, op_type='weighted_softmax_ce_loss',
+                          positive_cls_weight=positive_cls_weight, name='out')
+    return mx.sym.MakeLoss(model)
```

Comment:
As `weighted_softmax_ce_loss` doesn't use `out_grad` in `backward`, is `MakeLoss` needed here? I think this layer already acts like a loss layer. Please also see https://github.com/apache/incubator-mxnet/blob/master/example/numpy-ops/custom_softmax.py and  https://github.com/apache/incubator-mxnet/issues/8109.

Reply:
Not sure about that.. I got the following error if I remove make_loss:
```
mxnet.base.MXNetError: [17:57:50] src/executor/graph_executor.cc:98: Check failed: i < head_grads.size() && !head_grads[i].is_none() Because the last operator is not Loss function, head_gradient is required when calling backward. If you are attempting to minimize the output as an objective, please modify your network and pass it through the make_loss symbol.
```

Reply:
@eric-haibin-lin Please try `need_top_grad=False` as in https://github.com/apache/incubator-mxnet/blob/master/example/numpy-ops/custom_softmax.py#L45

Reply:
Yes, I tried this but this overrides the `kNullOp` req for data's gradient. I haven't spend much time on it yet.

Reply:
OK, I see. Thanks.